MaristCampusMap/MaristCampusMap/MainScreenViewController.swift

```diff
@@ -170,35 +178,32 @@ extension MainScreenViewController : UIPickerViewDataSource, UIPickerViewDelegat
     func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
         print(self.pickerOptions[row].name)
         self.selectedLocation = self.pickerOptions[row]
-        //self.SelectedLocationLabel.text = self.selectedLocation!.name
     }
 }
 
+
 extension MainScreenViewController : CLLocationManagerDelegate {
-    //print current location
+    //get user's current location
     func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
-        //let locValue:CLLocationCoordinate2D = manager.location!.coordinate
 
         if let location = locations.last {
             
             self.lastLocation = location
-            print(location)
-            //compare selected location to this location and get angle bearing for arrow
-            //then update arrow orientation, compare to self.selected
-            /*let _selectedLocation = CLLocation(latitude: selectedLocation.latitude, longitude: selectedLocation.longitude)
-            //print(location)
-            let bearingInRad = getBearingBetweenTwoPoints1(point1: location, point2: _selectedLocation)
             
-            print(rad2deg(bearingInRad))
-            //MapArrow.transform = CGAffineTransform(rotationAngle: CGFloat(bearingInRad))
-            let latestBearing = CGFloat(bearingInRad)
-            UIView.animate(withDuration: 0.5) {
-            self.MapArrow.transform = CGAffineTransform(rotationAngle: latestBearing - latestHeading)
-            }*/
-
+            if let selectedLocation =
+                self.selectedPickerOption?.location {
+            let _selectedLocation = CLLocation(latitude: selectedLocation.latitude, longitude: selectedLocation.longitude)
+            
+            let distanceInMeters = location.distance(from: _selectedLocation)
+            print(distanceInMeters)
+                if(distanceInMeters < 20.0) {
+                    ArrivedDisplay.isHidden = false
+                    MapArrow.isHidden = true
```

Comment:
Consider looking into using a single imageView component and simply changing the displayed image, rather than hiding one component and showing another.

Reply:
It is possible to change the image for the view. I tested it, the only issue is since the arrow rotates when I switch the image to arriveddisplay, that image is also rotating. I am going to look into a way to keep it fixed this week if I toggle between the images.